datameta/security/tfaz.py

```diff
@@ -0,0 +1,174 @@
+# Copyright 2021 Universität Tübingen, DKFZ and EMBL for the German Human Genome-Phenome Archive (GHGA)
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from datetime import datetime, timedelta
+import secrets
+
+from cryptography.fernet import Fernet
+import pyotp
+import qrcode
+import qrcode.image.svg
+from sqlalchemy import and_
+from pyramid import threadlocal
+from pyramid.settings import asbool
+
+from .tokenz import hash_token
+from ..models import User, Session, TfaToken
+
+import logging
+log = logging.getLogger(__name__)
+
+
+def is_2fa_enabled():
+    """Check if the system has 2fa enabled.
+
+    Returns:
+        - the 2fa activation status
+    """
+    return asbool(threadlocal.get_current_registry().settings['datameta.tfa.enabled'])
```

Comment:
Unfortunately, I don't know much about `threadlocal.get_current_registry()` and why it is used, but when I was looking earlier through the [documentation](https://docs.pylonsproject.org/projects/pyramid/en/latest/api/threadlocal.html), I saw that they state it should be used only in unit testing.

Reply:
This way of accessing this kind of variables is used throughout the datameta application (14 more mentions in 4 modules). If you know a proper way to access those, I guess we're open for suggestions :D

Reply:
Maybe [this ](https://stackoverflow.com/questions/10893628/how-can-i-get-the-ini-data-in-pyramid) is a solution, apparently you can also get the registry via the request. But I haven't tried it :D